src/Core/Domain/AuthorizationServer/Model/AuthorizedApplicationRepositoryInterface.php

```
@@ -0,0 +1,75 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Model;
+
+use PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Exception\ApplicationNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\ValueObject\ApplicationId;
+use PrestaShopBundle\Entity\AuthorizedApplication;
+
+interface AuthorizedApplicationRepositoryInterface
+{
+    /**
+     * @param AuthorizedApplication $application
+     *
+     * @return void
+     */
+    public function create(AuthorizedApplication $application): void;
+
+    /**
+     * @param AuthorizedApplication $application
+     *
+     * @return void
+     */
+    public function update(AuthorizedApplication $application): void;
+
+    /**
+     * @param ApplicationId $applicationId
+     *
+     * @throws ApplicationNotFoundException
+     *
+     * @return AuthorizedApplication|null
+     */
+    public function getById(ApplicationId $applicationId): ?AuthorizedApplication;
```

should return an exception if not found.

same for `getByName`